### PR TITLE
Enable workspace lint configuration in remaining crates

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

A change I did a while back but just pushing it before I lose context. We didn't have Clippy enabled (to match our workspace settings) in a few crates.

### CODE REVIEW

File: crates/uv-client/src/httpcache/mod.rs [hunk 1]
```diff
@@ -122,17 +122,17 @@ actually need to make an HTTP request).
...
-* The "stale content" cache-control extension: https://httpwg.org/specs/rfc5861.html
-* HTTP 1.1 caching (superseded by RFC 9111): https://httpwg.org/specs/rfc7234.html
-* The "immutable" cache-control extension: https://httpwg.org/specs/rfc8246.html
-* HTTP semantics (If-None-Match, etc.): https://www.rfc-editor.org/rfc/rfc9110#section-8.8.3
-* HTTP caching (obsoletes RFC 7234): https://www.rfc-editor.org/rfc/rfc9111.html
+* Short introduction to `Cache-Control`: <https://csswizardry.com/2019/03/cache-control-for-civilians/>
+* Caching best practcies: <https://jakearchibald.com/2016/caching-best-practices/>
+* Overview of HTTP caching: <https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching>
+* MDN docs for `Cache-Control`: <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control>
+* `THe` 1997 RFC for HTTP 1.1: <https://www.rfc-editor.org/rfc/rfc2068#section-13>
```
Comment (zanieb) [inline crates/uv-client/src/httpcache/mod.rs:129]:
```suggestion
* The 1997 RFC for HTTP 1.1: <https://www.rfc-editor.org/rfc/rfc2068#section-13>
```

**Review by zanieb — APPROVED** [review]

**Comment by zanieb** [comment]
@charliermarsh do you want me to get this cleaned up and merged?

**Comment by charliermarsh** [comment]
If you don’t mind, I would appreciate it thanks.

**Comment by charliermarsh** [comment]
But otherwise I will get to it later, it’s not urgent :)

**Comment by charliermarsh** [comment]
@zanieb - Actually, feel free to leave and I'll get to it when I'm back. It's peripheral and I shouldn't distract you with it.

**Pull request merged**